src/plugins/discourse/mirror.test.js

```diff
@@ -568,4 +654,548 @@ describe("plugins/discourse/mirror", () => {
+      const {mirror, fetcher, reporter, repo} = example();
+      fetcher.addPost(1, null);
+      fetcher.addLike("nope", 1, 1);
+      await mirror.update(reporter);
+      expect(repo.likes()).toEqual([]);
+    });
+
+    it("mirrors category definition topics only after it's interval", async () => {
+      const {mirror, fetcher, reporter, repo} = example();
+
+      fetcher.addCategory(4);
+      const topic4 = fetcher._categoryDefinitionTopic(4);
+      await mirror.update(reporter);
+      const firstTopics = repo.topics();
+
+      fetcher.addCategory(5);
+      await mirror.update(reporter);
+      const laterTopics = repo.topics();
+
+      expect(firstTopics).toEqual([topic4]);
+      expect(laterTopics).toEqual([topic4]);
+    });
+
+    it("mirrors respects recheckCategoryDefinitionsAfterMs option", async () => {
+      const {mirror, fetcher, reporter, repo} = example({
+        recheckCategoryDefinitionsAfterMs: 0,
+      });
+
+      fetcher.addCategory(4);
+      const topic4 = fetcher._categoryDefinitionTopic(4);
+      await mirror.update(reporter);
+      const firstTopics = repo.topics();
+
+      fetcher.addCategory(5);
+      const topic5 = fetcher._categoryDefinitionTopic(5);
+      await mirror.update(reporter);
+      const laterTopics = repo.topics();
+
+      expect(firstTopics).toEqual([topic4]);
+      expect(laterTopics).toEqual([topic4, topic5]);
+    });
+
+    it("mirrors respects recheckTopicsInCategories option", async () => {
+      const {mirror, fetcher, reporter, repo} = example({
+        recheckTopicsInCategories: [4],
+      });
+      const cooked = "<p>New content, without bump</p>";
+
+      // Adds a category description #4.
+      // And a topic 5 which is a member of the #4 category.
+      const p1 = fetcher.addPost(1, null);
+      fetcher.addPost(2, null);
+      fetcher.addPost(3, null);
+      fetcher.addCategory(4);
+      const p5 = fetcher.addPost(5, null);
+      fetcher.setTopicCategory(5, 4);
+
+      // See we're trying to update #4 and #5.
+      const topicsCall = jest.spyOn(fetcher, "topicWithPosts");
+      await mirror.update(reporter);
+
+      fetcher.editPost(p1, {cooked});
+      fetcher.editPost(p5, {cooked});
+      const earlyCalls = [...topicsCall.mock.calls];
+      topicsCall.mockClear();
+
+      await mirror.update(reporter);
+      const laterPosts = repo.posts();
+
+      expect(fetcher._topic(4)).toMatchObject({id: 4, categoryId: 4});
+      expect(fetcher._topic(5)).toMatchObject({id: 5, categoryId: 4});
+      expect(fetcher._post(p5)).toMatchObject({id: 5, cooked});
+      // The initial load has strict order requirements.
+      expect(earlyCalls).toEqual([[1], [2], [3], [5], [4]]);
+      // The update does not.
+      expect(topicsCall).toHaveBeenCalledTimes(2);
+      expect(topicsCall).toHaveBeenCalledWith(4);
+      expect(topicsCall).toHaveBeenCalledWith(5);
+      expect(laterPosts).toMatchInlineSnapshot(`
```

Comment:
Why do we keep this inline snapshot? Is it to verify that the updated version was retrieved? If so, wouldn't it be cleaner to just test those posts directly -- easier for people making future modifications to understand what the intended behavior is.

Reply:
That the update was retrieved for one, but not the other. Yes, I can update to directly testing if I keep a copy of pre-edit P1.

Reply:
Now uses `toEqual` with an array of posts.